### ISSUE TITLE
Azure.SQL.TDE is not required to enable IaC

### ISSUE THREAD

**Issue by BernieWhite** [Azure/PSRule.Rules.Azure#1530]
**Description of the issue**

New database already have TDE enabled so configuring the `Microsoft.Sql/servers/databases/transparentDataEncryption` sub-resource is not required, and can generate an error.

However, it can be disabled.

Also it should be checked for the _in-flight_ scenario.

**Expected behaviour**

Update rule to:

- Ensure TDE is not disabled.
- Check existence of the `Microsoft.Sql/servers/databases/transparentDataEncryption` sub-resource for exports only.

**Error output**

```text
InternalServerError: An unexpected error occured while processing the request.
```

**Module in use and version:**

- Module: PSRule.Rules.Azure
- Version: **1.17.1**
